Development mode for sync server (React Fast Refresh on port 5006)

lelemm opened the pull request:
I added a new command `yarn start:server-dev`: this command proxies calls to react from 5006 to port 3001 automatically. This way we can have the react fast refresh from the server port, not needing to connect from the 3001 port manually.

MikesGlitch commented on packages/sync-server/src/app.js:
```diff
@@ -67,9 +71,39 @@ app.use((req, res, next) => {
+      changeOrigin: true,
+      ws: true,
+      logLevel: 'debug',
+      context: (pathname, req) => {
```
I couldn't find any docs for this context 🤔

lelemm replied:
Yeah, that was a test before I understood I had to disable rateLimit. removed

MikesGlitch reviewed (commented):
Added some thoughts here: https://github.com/lelemm/actual/pull/3

Main changes were to: 
- Open the server url instead of the client url when launching
- Remove the config that wasn't working

Feel free to merge or to rework it but it would be good to have those things in this PR.

lelemm commented:
> Added some thoughts here: [lelemm#3](https://github.com/lelemm/actual/pull/3)
> 
> Main changes were to:
> 
> * Open the server url instead of the client url when launching
> * Remove the config that wasn't working
> 
> Feel free to merge or to rework it but it would be good to have those things in this PR.

I changed to open 5006 when running yarn start:server-dev
Also, I added a new command server-monitor that uses nodemon so changes on the server restarts it

lelemm commented:
saw your changes on your branch. better the `BROWSER_OPEN=localhost:5006` than changing vite config

MikesGlitch reviewed (commented):
Once the lint's fixed It'll be good to go

lelemm commented:
> Once the lint's fixed It'll be good to go

done

armanckeser commented:
I believe this PR broke my edge server:
`Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'http-proxy-middleware' imported from /app/src/app.js`

MikesGlitch commented:
> I believe this PR broke my edge server: `Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'http-proxy-middleware' imported from /app/src/app.js`

You'll need to run yarn install
